src/c_kzg_4844.c

```diff
@@ -990,14 +990,25 @@ C_KZG_RET blob_to_kzg_commitment(
     return C_KZG_OK;
 }
 
+/**
+ * Struct holding the argument to verify_kzg_proof_impl.
+ * Having a struct holding these allows some code simpliciations
+ *
+ * An instance of this struct models a claim (to be verified) that:
+ * proof_g1 is a proof for p(evaluation_point) == y_fr,
+ * where commitment_g1 is a commitment to the polynomial p.
+ */
+typedef struct {
+    g1_t commitment_g1;    /** Commitment to polynomial p*/
+    g1_t proof_g1;         /** Proof that p(evaluation_point) == y_fr*/
+    fr_t evaluation_point; /** point at which p is checked, typically determined
+                              as a hash output via Fiat-Shamir */
+    fr_t y_fr;             /** supposed value of p at the evaluation point */
+} kzg_evaluation_claim;
```

Comment:
When I see `kzg_evaluation_claim`, I don't know if it's a variable name or a type name.

It's idiomatic to make all the type names have the suffix of `*_t`, like fr_t, g1_t. So it's better to have the type name of `kzg_evaluation_claim_t` (if we think `kzg_evaluation_claim` is a good name)